src/components/Map.tsx

```diff
@@ -302,6 +333,34 @@ export class MapCanvas implements GestureTarget {
         this.dragLast = undefined;
     }
 
+    onMouseEnter(evt: PointerEvent) {
+        this.onMouseMove(evt);
+
+        this.redraw();
+    }
+
+    onMouseMove(evt: PointerEvent) {
+        const coord = clientCoord(evt);
+        const canvasCoords = this.clientToCanvas(coord);
+
+        this.mouseX = this.canvasToMap(canvasCoords.clientX - this.offsetX);
+        this.mouseY = this.canvasToMap(canvasCoords.clientY - this.offsetY);
+        this.canvasCoordinates = [this.mouseX,this.mouseY];
+
+        this.redraw();
```

Comment:
I don't think the map should redraw every time the mouse moves

Reply:
Agreed. This code was also in another PR, #70, which is also likely leading to all these merge conflicts. I'm not entirely sure how to optimize this, and it's my understanding that the way we're doing it now, we'd have to redraw at least part of the canvas. Can we merge this in for now and then create an issue so that this gets addressed for both this and #70?